go/vt/mysqlctl/reparent.go

```diff
@@ -91,16 +91,9 @@ func (mysqld *Mysqld) WaitForReparentJournal(ctx context.Context, timeCreatedNS
 }
 
 // DemoteMaster will gracefully demote a master mysql instance to read only.
-// If the master is still alive, then we need to demote it gracefully
-// make it read-only, flush the writes and get the position
+// By the time this method executes, the server should already have @@global.read_only=1
+// Thus, we should be able to get a consistent master position.
 func (mysqld *Mysqld) DemoteMaster() (rp mysql.Position, err error) {
```

Comment:
if you approve of this idea, we could even get rid of the `DemoteMaster` method and associated RPC calls, since this becomes just a wrapper around `MasterPosition`.

We can use the `MasterPosition` RPC definitions instead. I am not sure how to update the protobuf definitions though...